envi/archs/arm/disasm.py

```diff
@@ -1205,16 +1911,1816 @@ def p_uncond(opval, va):
+                    simdflags = (IFS_F64_U32, IFS_F64_S32)[signed]
+                    rbase1 = "d%d"
+                    rbase2 = "s%d"
+                else:
+                    d = (Vd<<1) | D
+                    simdflags = (IFS_F32_U32, IFS_F32_S32)[signed]
+                    rbase1 = "s%d"
+                    rbase2 = "s%d"
+
+            opers = (
+                    ArmRegOper(rctx.getRegisterIndex(rbase1%d)),
+                    ArmRegOper(rctx.getRegisterIndex(rbase2%m)),
+                    )
+
+        # VCVT p872
+        elif opc2 in (0b1010,0b1011,0b1110,0b1111) and opc3&1:
+            mnem = 'vcvt'
+            opcode = INS_VCVT
+
+            U = val1 & 1
+            imm4 = val2 & 0xf
+            sf = (val2>>8) & 1
+            sx = (val2>>7) & 1
+            i = (val2>>5) & 1
+
+            dp_op = sf
+
+            op = (val1>>2) & 1
+
+            Q = opc3
+
+            sfidx = (op << 3) | (sf << 2) | (sx << 1) | U
+            simdflags = (
+                    IFS_F32_S16, IFS_F32_U16, IFS_F32_S32, IFS_F32_U32,
+                    IFS_F64_S16, IFS_F64_U16, IFS_F64_S32, IFS_F64_U32,
+                    IFS_S16_F32, IFS_U16_F32, IFS_S32_F32, IFS_U32_F32,
+                    IFS_S16_F64, IFS_U16_F64, IFS_S32_F64, IFS_U32_F64,
+                    ) [sfidx]
+
+            size = (16, 32)[sx]
+            frac_bits = size - ((imm4<<1) | i)
+
+            if dp_op:
+                d = (D<<4) | Vd
+            else:
+                d = D | (Vd << 1)
+
+            rbase = ('s%d', 'd%d')[sf]
+
+            opers = (
+                    ArmRegOper(rctx.getRegisterIndex(rbase%d)),
+                    ArmRegOper(rctx.getRegisterIndex(rbase%d)),
+                    ArmImmOper(frac_bits),
+                    )
+
+    return (opcode, mnem, opers, iflags, simdflags)
+
+def p_advsimd_secondary(val, va, mnem, opcode, flags, opers):
+    if opcode == INS_VORR:
+        src1 = (val>>16) & 0xf
+        src2 = (val) & 0xf
+
+        if src1 == src2:
+            opers = (
+                ArmRegOper(rctx.getRegisterIndex(rbase%d)),
+                ArmRegOper(rctx.getRegisterIndex(rbase%n)),
+            )
+            return 'vmov', INS_VMOV, None, opers
+    return None, None, None, None
+
+adv_simd_dts = (IFS_S8, IFS_S16, IFS_S32, IFS_S64, 
+                IFS_U8, IFS_U16, IFS_U32, IFS_U64, 
+                IFS_I8, IFS_I16, IFS_I32, IFS_I64,
+                IFS_P8, IFS_P16, IFS_P32, IFS_P64,
+                IFS_F8, IFS_F16, IFS_F32, IFS_F64,
+                IFS_8,  IFS_16,  IFS_32,  IFS_64,
+                0,0,0,0,
+                IFS_F16_32, IFS_F32_16, 0,0,
+                IFS_F32_S32, IFS_F32_U32, IFS_S32_F32, IFS_U32_F32,
+                )
+ADV_SIMD_S8     = 0
+ADV_SIMD_U8     = 4
+ADV_SIMD_I8     = 8
+ADV_SIMD_P8     = 12
+ADV_SIMD_F8     = 16
+ADV_SIMD_8      = 20
+ADV_SIMD_NONE   = 24
+ADV_SIMD_F16F32 = 28
+ADV_SIMD_F32S32 = 32
+
+adv_simd_3_regs = (  # ABUC fields slammed together
```

Comment:
can i just say no?  this is the way the code made sense to write and to maintain.  and it would be a lot of work to revamp, and it would be more confusing to update and troubleshoot.  i don't think it can be pregen'd because of all the oddities.

Reply:
No is a perfectly valid answer :+1: